Sources/Purchasing/ProductsManager.swift

```diff
@@ -126,6 +135,25 @@ private extension ProductsManager {
         return self.productsFetcherSK1.products(withIdentifiers: identifiers, completion: completion)
     }
 
+    func trackProductRequestIfNeeded(_ startTime: Date,
+                                     storeKitVersion: StoreKitVersion,
+                                     error: PurchasesError?) {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *),
+           let diagnosticsTracker = self.diagnosticsTracker {
+            let requestTime = self.dateProvider.now().timeIntervalSince(startTime)
+            let errorMessage = (error?.userInfo[NSUnderlyingErrorKey] as? Error)?.localizedDescription
+                ?? error?.localizedDescription
+            let errorCode = error?.errorCode
+            Async.call(with: {}, asyncMethod: {
+                await diagnosticsTracker.trackProductsRequest(wasSuccessful: error == nil,
+                                                              storeKitVersion: storeKitVersion,
+                                                              errorMessage: errorMessage,
+                                                              errorCode: errorCode,
+                                                              responseTime: requestTime)
```

Comment:
Imo, either `responseTime` or `requestTime` should be renamed. What you're passing is actually the time the request took to come back right?

Reply:
Yeah that's true... but I was trying to reuse the same property we use for backend http request response times... I think I will rename the property to `responseTime` if that's ok, since I think it's ok to keep it as the same property (and so, reuse the validators in the backend)

Reply:
Addressed in https://github.com/RevenueCat/purchases-ios/pull/4247/commits/ea41bf72dc1b301b66a7cbdf02e1be8c01e3ccc8